mxnet-bot/LabelBotPredict/elastic_beanstalk_server/DataFetcher.py

```diff
@@ -0,0 +1,124 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This scipt is served to fetch GitHub issues into a json file
+from __future__ import print_function
+import os
+import requests
+import json
+import re
+import pandas as pd
+import logging
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+
+class DataFetcher:
+    GITHUB_USER = os.environ.get("GITHUB_USER")
+    GITHUB_OAUTH_TOKEN = os.environ.get("GITHUB_OAUTH_TOKEN")
+    REPO = os.environ.get("REPO")
+    assert GITHUB_USER and GITHUB_OAUTH_TOKEN and REPO, "Please set environment variables!"
+    REPO_URL = 'https://api.github.com/repos/%s/issues' % REPO
+    AUTH = (GITHUB_USER, GITHUB_OAUTH_TOKEN)
+
+    def __init__(self):
+        self.json_data = None
+
+    def cleanstr(self, string, substring):
+        # convert all non-alphanumeric charaters into substring
+        cleanstr = re.sub("[^0-9a-zA-Z]", substring, string)
+        return cleanstr.lower()
+
+    def count_pages(self, state):
+        # This method is to count how many pages of issues/labels in total
+        # state could be "open"/"closed"/"all", available to issues
+        response = requests.get(self.REPO_URL, {'state': state},
+                                auth=self.AUTH)
+        assert response.status_code == 200, "Authorization failed"
+        if "link" not in response.headers:
+            # That means only 1 page exits
+            return 1
+        # response.headers['link'] will looks like:
+        # <https://api.github.com/repositories/34864402/issues?state=all&page=387>; rel="last"
+        # In this case we need to extract '387' as the count of pages
+        # return the number of pages
+        return int(self.cleanstr(response.headers['link'], " ").split()[-3])
+    
+    def fetch_issues(self, numbers):
+        # number: a list of issue ids
```

Comment:
Can you change "numbers" to more understandable variable name?

Reply:
Done